numpy/core/setup.py

```diff
@@ -883,34 +872,39 @@ def generate_umath_c(ext, build_dir):
             join('src', 'umath', 'scalarmath.c.src'),
             join('src', 'umath', 'ufunc_type_resolution.c'),
             join('src', 'umath', 'override.c'),
-            join('src', 'private', 'mem_overlap.c'),
-            join('src', 'private', 'npy_longdouble.c'),
-            join('src', 'private', 'ufunc_override.c')]
+            # join('src', 'common', 'mem_overlap.c'),
+            # join('src', 'common', 'npy_longdouble.c'),
+            # join('src', 'common', 'ufunc_override.c'),
+            ]
 
     umath_deps = [
             generate_umath_py,
             join('include', 'numpy', 'npy_math.h'),
             join('include', 'numpy', 'halffloat.h'),
             join('src', 'multiarray', 'common.h'),
-            join('src', 'private', 'templ_common.h.src'),
+            join('src', 'common', 'templ_common.h.src'),
             join('src', 'umath', 'simd.inc.src'),
             join('src', 'umath', 'override.h'),
             join(codegen_dir, 'generate_ufunc_api.py'),
-            join('src', 'private', 'lowlevel_strided_loops.h'),
-            join('src', 'private', 'mem_overlap.h'),
-            join('src', 'private', 'npy_longdouble.h'),
-            join('src', 'private', 'ufunc_override.h'),
-            join('src', 'private', 'binop_override.h')] + npymath_sources
-
-    config.add_extension('umath',
-                         sources=umath_src +
+            join('src', 'common', 'lowlevel_strided_loops.h'),
+            join('src', 'common', 'mem_overlap.h'),
+            join('src', 'common', 'npy_longdouble.h'),
+            join('src', 'common', 'ufunc_override.h'),
+            join('src', 'common', 'binop_override.h')] + npymath_sources
+
+    config.add_extension('_multiarray_umath',
+                         sources=multiarray_src + umath_src +
```

Comment:
Might it be an idea to have a `common_src` here?

Reply:
adopted. Eventually everything will become common